File: backend/app/core/weeks.py
from datetime import date, timedelta


def monday_of(value: date) -> date:
    return value - timedelta(days=value.weekday())

def week_bounds(value: date) -> tuple[date, date]:
    start = monday_of(value)
    return start, start + timedelta(days=6)

def current_week_bounds(today: date | None = None) -> tuple[date, date]:
    return week_bounds(today or date.today())

File: frontend/src/lib/weeks.ts
import {
  addDays,
  addWeeks,
  format,
  parseISO,
  startOfWeek,
} from "date-fns";

export function mondayOf(value: Date): Date {
  return startOfWeek(value, { weekStartsOn: 1 });
}

export function weekBounds(value: Date): [Date, Date] {
  const start = mondayOf(value);
  return [start, addDays(start, 6)];
}

export function currentWeekBounds(): [Date, Date] {
  return weekBounds(new Date());
}

export function shiftWeek(start: Date, delta: number): [Date, Date] {
  return weekBounds(addWeeks(start, delta));
}

export function toApiDate(value: Date): string {
  return format(value, "yyyy-MM-dd");
}

export function formatWeekRange(
  start: string | Date,
  end: string | Date
): string {
  const from = typeof start === "string" ? parseISO(start) : start;
  const to = typeof end === "string" ? parseISO(end) : end;

  return `${format(from, "MMM d")} – ${format(to, "MMM d, yyyy")}`;
}
